added waypointAltitude if set

haaslukas opened the pull request:
in `terrainService.js` i saw that there is this TODO for the y-axis

```js
line.setAttribute("y1", "40"); // TODO: temp
line.setAttribute("y2", "40"); // TODO: temp
```

so i did a quick hack to get the waypoint altitudes if they are available. but needs testing on the server. i haven't had time yet to setup a proper local dev environment for this

haaslukas committed:
added waypointAltitude if set

opacopac commented:
hi @haaslukas, thanks a lot for your contribution for this often wanted feature!
i see one major issue with the provided solution: given not all waypoints of a route have an altitude set (e.g. in my own routes i only set the important ones) the line would jump up & down in an unrealistic way. also the waypoint altitude can be defined as "at the start/end of the leg", this should be considered and last but not least, the route should start & end on the ground level.
i will try to modify your solution to keep the previous value if no new altitude value is set instead of jumping back to the standard offset and also try to include the other two points mentioned above.

alexx-ftw commented:
@opacopac @haaslukas I am glad that this project is still being worked on. Thanks guys

haaslukas commented:
hi @opacopac 
you're absolutely right. sorry, should have seen that. that's the problem with not taking the right amount of time to prepare a dev server :see_no_evil:

opacopac commented:
hi @haaslukas
i understand that setting up a dev server can be quite a challenge, especially if there's a lot of inside knowledge required like here... i therefore created dockerfiles and added instructions for developers to the readme. **that way it should be easily possible to run it locally using docker**. maybe you like to give it a try.
unfortunately i had no time to work on a solution for the altitudes yet.

opacopac reviewed (changes requested):
like already mentioned in the comments, the following issues need to be addressed properly:
- given not all waypoints of a route have an altitude set, the line would jump up & down in an unrealistic way 
- the waypoint altitude can be defined as "at the start/end of the leg", this should be considered
- the route should start & end on the ground level

i imagine the solution to keep the previous value if no new altitude value is set instead of jumping back to the standard offset. additionally the solution should also work if none of the waypoints have an altitude set.

haaslukas committed:
Merge remote-tracking branch 'upstream/master'

haaslukas committed:
chore: ignore config.php and .claude/ for local dev

haaslukas commented:
Sorry for the long delay! Can confirm Docker setup works perfectly. Created a new PR #32 addressing the feedback here - smooth interpolation, ground level at start/end, and live terrain graph updates.
